### PR TITLE
Feature/FOUR-10790: Multiplayer support for boundary events

### PR DESCRIPTION
**Pull request by agustinbusso**
**Add support for Boundary events in collaborative modeler**

- Handle boundary creation.
- Handle position change when moving the parent task
- Handle position change when moving the boundary
- Handle boundary owner change (when moving a boundary from a task to another)

Related PRs
- [Modeler PR](https://github.com/ProcessMaker/modeler/pull/1689)
- [Realtime microservice](https://github.com/ProcessMaker/microservice-realtime/pull/11)

https://github.com/ProcessMaker/modeler/assets/90727999/eee16f33-c4a5-4496-88b9-f6a086dedde6

https://github.com/ProcessMaker/modeler/assets/90727999/14d6bd68-8378-4486-8e2f-26d22bd90edb

### CODE REVIEW

File: src/components/modeler/Selection.vue [hunk 1]
```diff
@@ -597,11 +602,45 @@ export default {
...
+      const boundaryEventTypes = [
+        boundaryErrorEventId,
+        boundaryConditionalEventId,
+        boundaryEscalationEventId,
+        boundaryMessageEventId,
+        boundarySignalEventId,
+        boundaryTimerEventId,
+      ];
+
+      const boundaryNodes = window.ProcessMaker.$modeler.nodes.filter(node => boundaryEventTypes.includes(node.type));
```
Comment (josechirivella) [inline src/components/modeler/Selection.vue:628]: Uhmm I'm pretty sure this nodes are available in the store?
Also, `window.ProcessMaker.$modeler` is a ref I believe also. Can you confirm @rodriquelca

File: src/components/nodes/boundaryEvent/boundaryEvent.vue [hunk 2]
```diff
@@ -202,6 +202,26 @@ export default {
...
     },
+    addMultiplayerBoundaryEvent() {
+      const control = {
+        bpmnType: this.node.diagram.$type,
+        id: this.node.diagram.id,
+        type: this.node.type,
+        attachedToRef: this.node.definition.get('attachedToRef'),
+      };
+
+      const { paper } = window.ProcessMaker.$modeler;
```
Comment (josechirivella) [inline src/components/nodes/boundaryEvent/boundaryEvent.vue:213]: same goes here, the paper should be available in the store or ref

File: src/multiplayer/multiplayer.js [hunk 3]
```diff
@@ -274,9 +280,32 @@ export default class Multiplayer {
...
+    if (element.component.node.definition.$type === 'bpmn:BoundaryEvent') {
+      this.attachBoundaryEventToNode(element, data);
+    }
+
     // Trigger a rendering of the element on the paper
     paper.findViewByModel(element).update();
   }
+  attachBoundaryEventToNode(element, data) {
+    const { paper } = this.modeler;
+    let node = this.getNodeById(data.attachedToRefId);
```
Comment (josechirivella) [inline src/multiplayer/multiplayer.js:293]:
```suggestion
    const node = this.getNodeById(data.attachedToRefId);
```

File: src/portsUtils.js [hunk 4]
```diff
@@ -4,6 +4,7 @@ import differenceWith from 'lodash/differenceWith';
 import isEqual from 'lodash/isEqual';
 
 function getModelPortPoints(model, group) {
+
```
Comment (josechirivella) [inline src/portsUtils.js:7]: remove

File: src/multiplayer/multiplayer.js [hunk 5]
```diff
@@ -165,11 +170,13 @@ export default class Multiplayer {
...
       'processmaker-modeler-data-input-association',
     ];
+
     return new Promise(resolve => {
       changes.map((data) => {
         if (flows.includes(data.type)) {
           this.createFlow(data);
-        } else {
+        }
+        else {
```
Comment (josechirivella) [inline src/multiplayer/multiplayer.js:179]:
```suggestion
        } else {
```

**Review by josechirivella — CHANGES_REQUESTED** [review]

**Review by rodriquelca — APPROVED** [review]

**Review by josechirivella — APPROVED** [review]

**Review by devmiguelangel — APPROVED** [review]

**Pull request merged**